### ISSUE TITLE
TypeError: exec_command() got an unexpected keyword argument 'sudoable'\n"

### ISSUE THREAD

**Issue by basictheprogram** [ansible/ansible#11275]
##### Issue Type:

Bug Report
##### Ansible Version:

$ ansible --version
ansible 2.0.0 (devel 5ed2e44026) last updated 2015/06/16 00:54:15 (GMT -500)
  lib/ansible/modules/core: (detached HEAD 9acc7c402f) last updated 2015/06/10 17:06:47 (GMT -500)
  lib/ansible/modules/extras: (detached HEAD 2f967a949f) last updated 2015/06/10 17:06:47 (GMT -500)
  v1/ansible/modules/core: (detached HEAD f8d8af17cd) last updated 2015/06/03 00:37:54 (GMT -500)
  v1/ansible/modules/extras: (detached HEAD 495ad450e5) last updated 2015/06/03 00:37:58 (GMT -500)
  configured module search path = None
##### Ansible Configuration:

I was (aka using v1 code base)
$ source v1/hacking/env-setup

Now I'm  (using v2 code base)
$ source hacking/env-setup
##### Environment:

OS X Yosemite 10.10.3

$ python -V
Python 2.7.6

$ pip freeze
ansible==2.0.0
ecdsa==0.13
epdb==0.12
httplib2==0.9.1
isodate==0.5.1
Jinja2==2.7.3
kerberos==1.2.2
MarkupSafe==0.23
nose==1.3.7
paramiko==1.15.2
passlib==1.6.2
pycrypto==2.6.1
pywinrm==0.0.3
PyYAML==3.11
six==1.9.0
xmltodict==0.9.2
##### Summary:

All my playbooks and plays that work in v1 fail in v2. I swear this was discovered when v2 dropped and there was a PR for it, but I cannot find the PR, the discussion, or issue. Vaguely recall it was with ssh_pass on OS X. Sorry
##### Steps To Reproduce:

$ source hacking/env-setup
ansible-playbook --ask-pass --ask-sudo-pass  -i production-servers linux-servers.yml -vvv
SSH password:
SUDO password[defaults to SSH password]:
##### Expected Results:

I assume the plays would run to completion like then do in v1.
##### Actual Results:

1 plays in linux-servers.yml
PLAY: ***************************************************************************
TASK [setup] ********************************************************************
fatal: [host.domain.com]: FAILED! => {"failed": true, "exception": "Traceback (most recent call last):\n  File \"/Users/tanner/projects/ansible.git/ansible/lib/ansible/executor/process/worker.py\", line 118, in run\n    executor_result = TaskExecutor(host, task, job_vars, new_connection_info, self._new_stdin, self._loader, shared_loader_obj).run()\n  File \"/Users/tanner/projects/ansible.git/ansible/lib/ansible/executor/task_executor.py\", line 111, in run\n    res = self._execute()\n  File \"/Users/tanner/projects/ansible.git/ansible/lib/ansible/executor/task_executor.py\", line 266, in _execute\n    result = self._handler.run(task_vars=variables)\n  File \"/Users/tanner/projects/ansible.git/ansible/lib/ansible/plugins/action/normal.py\", line 27, in run\n    return self._execute_module(tmp, task_vars=task_vars)\n  File \"/Users/tanner/projects/ansible.git/ansible/lib/ansible/plugins/action/__init__.py\", line 348, in _execute_module\n    tmp = self._make_tmp_path()\n  File \"/Users/tanner/projects/ansible.git/ansible/lib/ansible/plugins/action/__init__.py\", line 156, in _make_tmp_path\n    result = self._low_level_execute_command(cmd, None, sudoable=False)\n  File \"/Users/tanner/projects/ansible.git/ansible/lib/ansible/plugins/action/__init__.py\", line 449, in _low_level_execute_command\n    rc, stdin, stdout, stderr = self._connection.exec_command(cmd, tmp, in_data=in_data, sudoable=sudoable)\nTypeError: exec_command() got an unexpected keyword argument 'sudoable'\n", "stdout": ""}

NO MORE HOSTS LEFT **************************************************************

PLAY RECAP **********************************************************************
host.domain.com             : ok=0    changed=0    unreachable=0    failed=1

**Comment by bennojoy** [comment]
@basictheprogram  which connection type are you using ? seems like  an older plugin is getting called here.

**Comment by basictheprogram** [comment]
v1's paramiko_ssh.py (which I believe OS X must user because of the ssh_pass issue)

``` python
def exec_command(self, cmd, tmp_path, become_user=None, sudoable=False, executable='/bin/sh', in_data=None):
```

v2 paramiko_ssh.py

``` python
def exec_command(self, cmd, tmp_path, executable='/bin/sh', in_data=None):
```

Which explains the unexpected keyword argument?

**Comment by basictheprogram** [comment]
I did not change my default connection type so ssh? But with OS X is it not forced to paramiko? So I think the correct full answer would be ssh --ask-pass --ask-sudo-pass?

**Comment by bennojoy** [comment]
@basictheprogram  yeah iirc the connection by default is made paramiko in osx, seems like a bug where paramiko will check, can you please try if -c ssh works.

**Comment by basictheprogram** [comment]
I assume just add -c ssh to my normal invocation? 

fatal: [host.domain.com]: FAILED! => {"msg": "ERROR! to use the 'ssh' connection type with passwords, you must install the sshpass program", "failed": true}

https://gist.github.com/arunoda/7790979

Installing on OS X is tricky, since there is no official build for it. Before you get started, you need install xcode and command line tools.

Adding a lol....

$ brew install sshpass
Error: No available formula for sshpass
We won't add sshpass because it makes it too easy for novice SSH users to
ruin SSH's security.

**Comment by bcoca** [comment]
This is me revamping the connection plugins with become support, i did ssh and am in the middle of local, I have not gotten to paramiko yet (will do soon).

**Comment by bcoca** [comment]
made small fix 98f5534d9c08950ca60afecf4e1725459431d551 to get you working, try refreshing from git and see if it works
